src/plugins/orangehrmAdminPlugin/Service/LocalizationService.php

```diff
@@ -270,4 +276,122 @@ private function getXliffXmlSources(array $i18nGroups, I18NLanguage $language):
         }
         return $xml;
     }
+
+    public function symfonyXliffValidations(string $content, ?string $file = null): array
+    {
+        $errors = [];
+        // Avoid: Warning DOMDocument::loadXML(): Empty string supplied as input
+        if ('' === trim($content)) {
+            return ['file' => $file, 'valid' => true];
+        }
+
+        $internal = libxml_use_internal_errors(true);
+
+        $document = new \DOMDocument();
+        $document->loadXML($content);
+
+        if (null !== $targetLanguage = $this->getTargetLanguageFromFile($document)) {
+            $normalizedLocalePattern = sprintf(
+                '(%s|%s)',
+                preg_quote($targetLanguage, '/'),
+                preg_quote(str_replace('-', '_', $targetLanguage), '/')
+            );
+            // strict file names require translation files to be named '____.locale.xlf'
+            $expectedFilenamePattern = sprintf('/^.*\.(?i:%s)\.(?:xlf|xliff)/', $normalizedLocalePattern);
+
+            if (0 === preg_match($expectedFilenamePattern, basename($file))) {
+                $errors[] = [
+                    'line' => -1,
+                    'column' => -1,
+                    'message' => sprintf(
+                        'There is a mismatch between the language included in the file name ("%s") and the "%s" value used in the "target-language" attribute of the file.',
+                        basename($file),
+                        $targetLanguage
+                    ),
+                ];
+            }
+        }
+
+        foreach (XliffUtils::validateSchema($document) as $xmlError) {
+            $errors[] = [
+                'line' => $xmlError['line'],
+                'column' => $xmlError['column'],
+                'message' => $xmlError['message'],
+            ];
+        }
+
+        libxml_clear_errors();
+        libxml_use_internal_errors($internal);
+
+        return ['file' => $file, 'isValid' => 0 === \count($errors), 'messages' => $errors];
```

Comment:
just noting down that code taken from `XliffLintCommand` in `symfony/translations`
https://github.com/symfony/translation/blob/5.4/Command/XliffLintCommand.php#L123-L156

We'll have to discuss this a bit. Not sure if we need the filename checking part. 

Better to create a utility class for `XlfifUtils` for now. Reference: https://github.com/orangehrm/orangehrm/blob/main/src/plugins/orangehrmCorePlugin/Utility/Sanitizer.php